src/gui/screen_filebrowser.cpp

```diff
@@ -101,57 +101,55 @@ static int screen_filebrowser_event(screen_t *screen, window_t *window,
         return 0;
     }
 
-    const FILINFO *file_item = &(filelist->file_items[(intptr_t)param]);
+    // const FILINFO *file_item = &(filelist->file_items[(intptr_t)param]);
+
+    bool currentIsFile;
+    // displayed text - can be a 8.3 DOS name or a LFN
+    const char *currentFName = window_file_current_fname(filelist, &currentIsFile);
+
     //there must be fname, not altname
-    if (!strcmp(file_item->fname, "..") && !strcmp(filelist->altpath, "/")) {
+    if (!strcmp(currentFName, "..") && !strcmp(filelist->altpath, "/")) {
         screen_close();
         return 1;
     }
 
-    if (file_item->fattrib & AM_DIR) {        // directory selected
-        if (strcmp(file_item->fname, "..")) { // not same -> not ..
-            if ((strlen(filelist->altpath) + strlen(file_item->altname) + 1) >= MAXPATHNAMELENGTH) {
-                LOG_ERROR("path too long");
-                return 0;
-            }
-
-            size_t len = strlen(filelist->altpath);
-            if (filelist->altpath[len - 1] != '/') {
-                filelist->altpath[len++] = '/';
+    size_t altPathLen = strlen(filelist->altpath);
+    if ((altPathLen + strlen(currentFName) + 1) >= MAXPATHNAMELENGTH) {
```

Comment:
What is the difference between `MAXPATHNAMELENGTH` and `F_ MAXPATHNAMELENGTH` used in the header?

Reply:
Honestly, I have no idea what were the original intentions, this has been there already.
I'll make it more consistent...